cmd/snap/cmd_snap_op_test.go

```diff
@@ -135,7 +135,7 @@ func (s *SnapOpSuite) TestWait(c *check.C) {
 	// lazy way of getting a URL that won't work nor break stuff
 	server := httptest.NewServer(nil)
 	snap.ClientConfig.BaseURL = server.URL
-	server.Close()
+	defer server.Close()
```

Comment:
Is there not a better way of getting a URL?

Reply:
hm this changes the test slightly, previously we were getting:
```
cannot communicate with server: Get "http://127.0.0.1:35035/v2/changes/x": dial tcp 127.0.0.1:35035: connect: connection refused
```
which I think was the point of the test, and now it's:
```
cannot decode " page not found\n": json: cannot unmarshal number into Go value of type client.response
```

there's a similar problem in line 217.

What's the real benefit of using httptest.NewServer() here? The idea was to get a valid base URL I believe, with a port chosen seemingly at random (which we've already established is not the case). What if we simply used:
```
snap.ClientConfig.BaseURL = "http://127.0.0.1:61010/"
```
or some other distinct fixed port number in the 60k range? I think it's no worse than the original code which was already prone to races and resulted in hard to track failures.

Reply:
Doesn't picking a random fixed port have the same problem that we have right now (that it could actually be used by some other test)? Or is there something special about ports in the 60000-65535 range?

Reply:
yes, but at least we can easily identify the problem should it arise, also I think it's unlikely to conflict with we use different fixed number in those 2 tests in question

Reply:
What about setting the URL to `http://localhost:0`. From my understanding, 0 isn't a valid port number, and will always result in a connection refused error:
```
cannot communicate with server: Get "http://localhost:0/v2/changes/x": dial tcp 127.0.0.1:0: connect: connection refused
```